File: SIMINOV-CONNECT-TEMPLATE/src/siminov/connect/template/reader/LiquorsReader.java
package siminov.connect.template.reader;

import java.io.InputStream;
import java.util.ArrayList;
import java.util.Collection;
import java.util.Iterator;

import org.xml.sax.Attributes;
import org.xml.sax.SAXException;

import siminov.connect.template.Constants;
import siminov.connect.template.model.Liquor;
import siminov.orm.exception.DeploymentException;
import siminov.orm.log.Log;
import siminov.orm.reader.SiminovSAXDefaultHandler;

public class LiquorsReader extends SiminovSAXDefaultHandler implements Constants {

	private Collection<Liquor> liquors = new ArrayList<Liquor>();
	private Liquor liquor = null;
	
	
	private StringBuilder tempValue = new StringBuilder();

	
	public LiquorsReader(InputStream data) {
		
		if(data == null) {
			return;
		}
		
		
		try {
			parseMessage(data);
		} catch(Exception exception) {
			Log.loge(getClass().getName(), "Constructor", "Exception caught while parsing Liquors, " + exception.getMessage());
			throw new DeploymentException(getClass().getName(), "Constructor", "Exception caught while parsing Liquors, " + exception.getMessage());
		}
	}

	public void startElement(String uri, String localName, String qName, Attributes attributes) throws SAXException {
		super.startElement(uri, localName, qName, attributes);
		
		tempValue = new StringBuilder();
		
		if(localName.equalsIgnoreCase(GET_LIQUORS_WS_LIQUOR)) {
			liquor = new Liquor();
		}
	}
	
	public void characters(char[] ch, int start, int length) throws SAXException {
		super.characters(ch, start, length);
		
		String value = new String(ch,start,length);
		
		if(value == null || value.length() <= 0 || value.equalsIgnoreCase(NEW_LINE)) {
			return;
		}
		
		value = value.trim();
		tempValue.append(value);
	}

	public void endElement(String uri, String localName, String qName) throws SAXException {
		super.endElement(uri, localName, qName);
		
		if(localName.equalsIgnoreCase(GET_LIQUORS_WS_LIQUOR)) {
			liquors.add(liquor);
		} else if(localName.equalsIgnoreCase(GET_LIQUORS_WS_LIQUOR_NAME)) {
			liquor.setLiquorType(tempValue.toString());
		} else if(localName.equalsIgnoreCase(GET_LIQUORS_WS_LIQUOR_DESCRIPTION)) {
			liquor.setDescription(tempValue.toString());
		} else if(localName.equalsIgnoreCase(GET_LIQUORS_WS_LIQUOR_HISTROY)) {
			liquor.setHistory(tempValue.toString());
		} else if(localName.equalsIgnoreCase(GET_LIQUORS_WS_LIQUOR_LINK)) {
			liquor.setLink(tempValue.toString());
		} else if(localName.equalsIgnoreCase(GET_LIQUORS_WS_LIQUOR_ALCHOL_CONTENT)) {
			liquor.setAlcholContent(tempValue.toString());
		}
	}
	
	public Iterator<Liquor> getLiquors() {
		return this.liquors.iterator();
	}
}

File: Connect-Sample/iOS/Siminov/Connect/Sample/Reader/LiquorsReader.m
//
//  LiquorsReader.m
//  connect-sample
//
//  Created by Geetika on 29/05/15.
//  Copyright (c) 2015 Siminov. All rights reserved.
//

#import "LiquorsReader.h"
#import "Constants.h"

@implementation LiquorsReader

- (id)initWithData:(NSData * const)data {
    
    liquors = [[NSMutableArray alloc]init];
    tempValue = [[NSMutableString alloc] init];
    
    if (data == nil) {
        return self;
    }
    
    @try {
        [self parseMessage:data];
    }
    @catch (NSException *exception) {
        [SICLog error:NSStringFromClass([self class]) methodName:@"Constructor" message:[NSString stringWithFormat:@"Exception caught while parsing Liquors, %@",[exception reason]]];
        @throw [[SICDeploymentException alloc]initWithClassName:NSStringFromClass([self class]) methodName:@"Constructor" message:[NSString stringWithFormat:@"Exception caught while parsing Liquors, %@",[exception reason]]];
    }
    return self;
}

- (void)parser:(NSXMLParser *)parser didStartElement:(NSString *)elementName namespaceURI:(NSString *)namespaceURI qualifiedName:(NSString *)qName attributes:(NSDictionary *)attributeDict {
    
    tempValue = [[NSMutableString alloc] init];
    
    if([elementName caseInsensitiveCompare:GET_LIQUORS_WS_LIQUOR] == NSOrderedSame) {
        liquor = [[Liquor alloc] init];
    }
}

- (void)parser:(NSXMLParser *)parser foundCharacters:(NSString *)string {
    tempValue = (NSMutableString *)[string stringByTrimmingCharactersInSet:[NSCharacterSet whitespaceAndNewlineCharacterSet]];
}

- (void)parser:(NSXMLParser *)parser didEndElement:(NSString *)elementName namespaceURI:(NSString *)namespaceURI qualifiedName:(NSString *)qName {
    
    if([elementName caseInsensitiveCompare:GET_LIQUORS_WS_LIQUOR] == NSOrderedSame) {
        [liquors addObject:liquor];
    } else if([elementName caseInsensitiveCompare:GET_LIQUORS_WS_LIQUOR_NAME] == NSOrderedSame) {
        [liquor setLiquorType:(NSString *)tempValue];
    } else if([elementName caseInsensitiveCompare:GET_LIQUORS_WS_LIQUOR_DESCRIPTION] == NSOrderedSame) {
        [liquor setDescription:(NSString *)tempValue];
    } else if([elementName caseInsensitiveCompare:GET_LIQUORS_WS_LIQUOR_HISTROY] == NSOrderedSame) {
        [liquor setHistory:(NSString *)tempValue];
    } else if([elementName caseInsensitiveCompare:GET_LIQUORS_WS_LIQUOR_LINK] == NSOrderedSame) {
        [liquor setLink:(NSString *)tempValue];
    } else if([elementName caseInsensitiveCompare:GET_LIQUORS_WS_LIQUOR_ALCHOL_CONTENT] == NSOrderedSame) {
        [liquor setAlcholContent:(NSString *)tempValue];
    }
}

- (NSEnumerator *)getLiquors {
    return [liquors objectEnumerator];
}

@end
